lib/plug_caisson.ex

```diff
@@ -89,6 +89,9 @@ defmodule PlugCaisson do
           _ ->
```

Comment:
small nit: this entire `fetch_decompressor/2` block should be refactored to a `with ... do...else` with the `content_encoding` case as the happy path and all other situations under `else`. The 3 level nesting affects readability quite a bit.